### DISCUSSION TITLE
MoviePy Effects and Transitions like CapCut

### DISCUSSION THREAD

**Discussion by thprtm** [Zulko/moviepy#2119] [General]
Hi, ive seen such types of videos, is this kind of transitions and effects be done with moviepy? It seems so handmade in CapCut.

https://github.com/Zulko/moviepy/assets/149414595/342d1afd-cef1-447e-bd98-e47d43bd13b5

**Comment by GentlemanHu** [comment]
+1

Maybe remotion can do it.
https://github.com/remotion-dev/remotion

**Comment by usamaehsan** [comment]
@thprtm have you found anything?

**Comment by Implosiv3** [comment]
I'm building an engine on top of moviepy able to do that and yes, moviepy can do almost everything you can think about. Videos are only matrixes of pixels.

You just need to imagine the scene in your mind and tell it to the code :). For example, in this video below everything has been done with moviepy. The Google Search has been generated with only an image, and has been placed inside a greenscreen also using moviepy. Then, I put some transitions between clips, and even I put one clip moving from left to right on top of another clip (and also, if you pay attention, the sound of the moving clip is moving also from left to right).

Moviepy has an incredible power, trust me. Of course, this doesn't exist as a premade by default, you need to build your own transition generator by code and use it properly, but it is possible to automate it 👇

https://github.com/user-attachments/assets/e930164c-d24b-4df7-a03e-7abdb0f1faca

**Comment by krishnabisw** [comment]
[Capcut Pro](https://capkcutapk.com/) is a popular apk video editing app, with this apps you can edit many types of videos, it's a really good app to edit videos. With the help of this app you can edit all types of videos like TikTok, YouTube, Instagram Reels etc.

**Comment by TheStarkster** [comment]
@thprtm
I am able to do this using shaders though, and going to launch a product soon

https://github.com/user-attachments/assets/d7635f00-428a-421d-a409-3083622160df
